js/piwik.js

```diff
@@ -3869,47 +3869,134 @@ if (typeof Piwik !== 'object') {
-                        processClick(target);
-                    }
-                } else if (evt.type === 'mousedown') {
-                    if ((button === 1 || button === 2) && target) {
-                        lastButton = button;
-                        lastTarget = target;
-                    } else {
+                    event = event || windowAlias.event;
+
+                    var button = getNameOfClickedButton(event);
+                    var target = getTargetElementFromEvent(event);
+
+                    if (event.type === 'click') {
+
+                        var ignoreClick = false;
+                        if (enable && button === 'middle') {
+                            // if enabled, we track middle clicks via mouseup
+                            // some browsers (eg chrome) trigger click and mousedown/up events when middle is clicked,
+                            // whereas some do not. This way we make "sure" to track them only once, either in click
+                            // (default) or in mouseup (if enable == true)
+                            ignoreClick = true;
+                        }
+
+                        if (target && !ignoreClick) {
+                            processClick(target);
+                        }
+                    } else if (event.type === 'mousedown') {
+                        if (button === 'middle' && target) {
+                            lastButton = button;
+                            lastTarget = target;
+                        } else {
+                            lastButton = lastTarget = null;
+                        }
+                    } else if (event.type === 'mouseup') {
+                        if (button === lastButton && target === lastTarget) {
+                            processClick(target);
+                        }
                         lastButton = lastTarget = null;
-                    }
-                } else if (evt.type === 'mouseup') {
-                    if (button === lastButton && target === lastTarget) {
+                    } else if (event.type === 'contextmenu') {
                         processClick(target);
                     }
-                    lastButton = lastTarget = null;
-                }
+                };
             }
 
             /*
              * Add click listener to a DOM element
              */
             function addClickListener(element, enable) {
+                addEventListener(element, 'click', clickHandler(enable), false);
+
                 if (enable) {
-                    // for simplicity and performance, we ignore drag events
-                    addEventListener(element, 'mouseup', clickHandler, false);
-                    addEventListener(element, 'mousedown', clickHandler, false);
-                } else {
-                    addEventListener(element, 'click', clickHandler, false);
+                    addEventListener(element, 'mouseup', clickHandler(enable), false);
```

Comment:
We now use the `mouseup/mousedown` event specifically to only track middle clicks. This was not the case before and I think we did not track some outlinks/downloads on older IE versions but not sure.